libp2p/network/url_conversion_test.go

```diff
@@ -105,14 +104,35 @@ func TestPeerMultiAddrFromServiceURL_Error(t *testing.T) {
 			t.Run(testName, func(t *testing.T) {
 				serviceURL := fmt.Sprintf(testCase.serviceURLFormat, hostname)
 				actualMultiaddr, err := Libp2pMultiaddrFromServiceURL(serviceURL)
-				// TECHDEBT: assert specific errors
-				// Print resulting multiaddr to understand why no error.
-				require.ErrorContainsf(t, err, testCase.expectedErrContains, fmt.Sprintf("actualMultiaddr: %s", actualMultiaddr))
+				assert.Error(t, err, fmt.Sprintf("actualMultiaddr: %s", actualMultiaddr))
+				assert.True(t, strings.HasPrefix(err.Error(), testCase.expectedErr), fmt.Sprintf("expected error to start with %q, but got %q", testCase.expectedErr, err.Error()))
```

Comment:
Is there a specific reason you moved away from `#ErrorContainsf`? In my estimation, it's a simplification of the combination of `#Error` and `#True` with `strings.HasPrefix` like you've done here. I don't have a strong opinion about the message that the assertion prints when it fails, the only reason I had it print the `actualMultiaddr` was for debugging. If that assertion failed, I wanted to know what the result was but that can be an exercise left to the developer if this test starts failing. I believe the test harness will print the difference between the actual and expectation so you don't have to `fmt` if yourself.

I would also prefer `require` over `assert`, generally, to avoid unnecessarily verbose failure output. In this case it's not a big deal but I think we should be consistent unless there's a principled reason to make an exception (in which case, I would encourage a comment.